edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -1272,13 +1276,53 @@ fn get_notary_parameters(
     Some((notary_auth, gun, tag, config_path.to_path_buf()))
 }
 
+// Disallow adding privileged and other capabilities if allow_privileged is false
+fn unset_privileged(allow_privileged: bool, create_options: &mut ContainerCreateBody) {
+    if let Some(config) = create_options.host_config() {
+        if !allow_privileged && (config.privileged() == Some(&true) || !config.cap_add().is_empty())
+        {
+            warn!("Privileged capabilities are disallowed on this device. Privileged capabilities can be used to gain root access. If a module needs to run as privileged, and you are aware of the consequences, set `allow_privileged` to `true` in the config.toml and restart the service.");
+            let mut config = config.clone();
```

Comment:
Can you just do `if let Some(mut config)` on 1281? Then you don't need this.

Reply:
Nope, `create_options.host_config()` returns a non-mutable reference to the host config. There is no mutable getter by design. I think its fine to keep it this way and continue using the get/set pattern.